crates/sui-core/src/authority/authority_store.rs

```diff
@@ -1124,25 +1137,25 @@ impl AuthorityStore {
         }
     }
 
-    pub fn get_certified_transaction(
+    pub fn get_executed_transaction(
         &self,
-        transaction_digest: &TransactionDigest,
-    ) -> SuiResult<Option<VerifiedCertificate>> {
-        let transaction = self.perpetual_tables.certificates.get(transaction_digest)?;
-        Ok(transaction.map(|t| t.into()))
+        tx_digest: &TransactionDigest,
+    ) -> Result<Option<SenderSignedData>, TypedStoreError> {
+        self.perpetual_tables.executed_transactions.get(tx_digest)
     }
 
-    pub fn multi_get_certified_transaction(
+    pub fn get_certified_transaction(
         &self,
-        transaction_digests: &[TransactionDigest],
-    ) -> SuiResult<Vec<Option<VerifiedCertificate>>> {
-        Ok(self
-            .perpetual_tables
-            .certificates
-            .multi_get(transaction_digests)?
-            .into_iter()
-            .map(|o| o.map(|c| c.into()))
-            .collect())
+        tx_digest: &TransactionDigest,
+    ) -> Result<Option<VerifiedCertificate>, TypedStoreError> {
+        let transaction = self.get_executed_transaction(tx_digest)?;
+        let cert_sig = self.get_transaction_cert_sig(tx_digest)?;
```

Comment:
I'd personally prefer a fail fast solution here; if the first response is None then return, no need to run the second DBget.